### PR TITLE
ref(rust): Add --python-max-queue-depth option

### PR DESCRIPTION
**Pull request by untitaker**
The python message processor is not using all cores. Up until now we
thought that was because check_for_results blocks too often, and tried
to fix that in https://github.com/getsentry/snuba/pull/4703, reverted in
https://github.com/getsentry/snuba/pull/4757. But perhaps all we have to
do is to raise MessageRejected less often?

### CODE REVIEW

File: snuba/cli/rust_consumer.py [hunk 1]
```diff
@@ -106,6 +106,12 @@
     default=None,
     help="Kafka group instance id. passing a value here will run kafka with static membership.",
 )
+@click.option(
+    "--python-max-queue-depth",
+    type=int,
+    default=None,
+    help="How many messages should be queued up before the Python message processor before backpressure kicks in. Defaults to the number of processes.",
```
Comment (loewenheim) [inline snuba/cli/rust_consumer.py:113]: This reads wrong to me, but maybe I'm misunderstanding it?

Comment (untitaker) [reply snuba/cli/rust_consumer.py:113]: the default or which part?

Comment (loewenheim) [reply snuba/cli/rust_consumer.py:113]: "queued up before the Python message processor before backpressure kicks in."

Comment (loewenheim) [reply snuba/cli/rust_consumer.py:113]: I think both of those would be improvements

**Review by lynnagara — APPROVED** [review]
let's give it a shot!

**Review by Swatinem — APPROVED** [review]

**Pull request merged**